src/big_query/big_query_table_metadata.py

```diff
@@ -16,6 +19,40 @@ def table_exists(self):
             return False
         return True
 
+    @staticmethod
+    def __get_table_or_partition(project_id, dataset_id, table_id,
+                                 partition_id):
+        table_metadata = BigQuery().get_table(project_id, dataset_id,
+                                              BigQueryTableMetadata.get_table_id_with_partition_id(
+                                                  table_id, partition_id))
+        return BigQueryTableMetadata(table_metadata)
+
+    @staticmethod
+    def get_table_id_with_partition_id(table_id, partition_id):
+        return table_id + ('' if partition_id is None else '$' + partition_id)
+
+    @staticmethod
+    def get_table_by_reference(reference):
+        return BigQueryTableMetadata.__get_table_or_partition(project_id=reference.project_id,
+                                                              dataset_id=reference.dataset_id,
+                                                              table_id=reference.table_id,
+                                                              partition_id=reference.partition_id)
+
+    @staticmethod
+    @cached(time=300)
+    def get_table_by_reference_cached(reference):
+        return BigQueryTableMetadata.get_table_by_reference(reference)
+
+    def create_the_same_empty_table(self, target_reference):
+        body = {}
```

Comment:
Maybe more pythonic way like in https://github.com/ocadotechnology/gcp-census/blob/master/gcp_census/bigquery/bigquery_client.py#L141-L146

Reply:
Done